### DISCUSSION TITLE
如何将 org-media-note 的时间戳，时间段导出成原地址

### DISCUSSION THREAD

**Discussion by czqhurricane** [yuchen-lea/org-media-note#52] [Q&A] [answered]
[[video:https://www.bilibili.com/video/BV1th411z7sn/?p=14&vd_source=ac634ae692c093e75476756c65a3382b#0:15:37-0:17:30][0:15:37-0:17:30]]

org-media-note 的时间戳，时间段的格式是 url#t1 或者 url#t1-t2。
并且 t 的格式不是按秒计算的。

我发现油管，B 站的有时间的链接都是：

```
https://www.youtube.com/live/L34xplC3qKU?si=cyJyJK_6LhKOq86u&t=1852

【STM32入门教程-2023版 细致讲解 中文字幕】 【精准空降到 05:31】 https://www.bilibili.com/video/BV1th411z7sn/?p=3&share_source=copy_web&t=331

```
url&t=... 的格式。

我如果将 org 文件导出成 html，希望变成如上格式，这样在浏览器点击就能跳转到时间点。

该如何实现？

**Comment by yuchen-lea** [comment]
1.  需要使用`org-link-set-parameters`来设置链接的`:export`函数

2. 在导出函数中，处理url#t1 或者 url#t1-t2，有函数 `org-media-note--timestamp-to-seconds` 用来把timestamp转化为秒，你可能需要考虑舍弃毫秒

你可以先试试，如果比较通用可以合到repo里。似乎youtube和bilibili都是这种格式，不知道其他网站。此外，本地的链接导出要怎么处理我还没什么想法。

**Comment by yuchen-lea** [comment, marked as answer]
d1363ee 先简单实现了一版：仅支持html导出时bilibili和youtube的空降链接，可以用 `org-media-note-desc-fn` 自定义链接样式，默认为timestamp

如果有其他跟导出相关的需求，欢迎继续讨论

**Reply by czqhurricane** [reply]
https://github.com/yuchen-lea/org-media-note/blob/d1363ee32326c8e9a5ae5f7b4c95d405b54bb909/org-media-note.el#L197-L214

中的

https://github.com/yuchen-lea/org-media-note/blob/d1363ee32326c8e9a5ae5f7b4c95d405b54bb909/org-media-note.el#L207

我这里需要 B 站链接改成 &，才能正确跳转，否则是乱码。B，y，两个的2时间戳都是用 & 符号分隔的啊。为什么这里判断 B 是用的问号。？

**Reply by yuchen-lea** [reply]
把你的链接贴出来，我这边测试的结果如下：

可以打开：
    - https://www.bilibili.com/video/BV1Q4411K7MX/?t=253
    - https://www.bilibili.com/video/BV1Q4411K7MX?t=253

? 改成 & 是无效网址。

而且我试过有一堆参数的 https://www.bilibili.com/video/BV13X4y1n7gC/?share_source=copy_web&vd_source=f7eac721ceb9dca4eea0199cb8e89222?t=97  也可以正常打开
